src/rag/reflexion_engine.py

```
@@ -537,13 +585,46 @@ async def query_with_reflexion_stream(
+                                -1
+                            ].evaluation.confidence_score
+                            if reflexion_memory.cycles
+                            else 0.0,
+                            "web_search_used": any(
+                                cycle.web_search_enabled
+                                for cycle in reflexion_memory.cycles
+                            ),
+                        },
+                    )
+                    logger.info("Stored QA cache entry for future similar questions")
+                except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)